// seed_data/objects/courses/fast_track_business_success/foundation/_ftbsod_en_default_email_series.js
var ReferenceTo = require('seed_data/ReferenceTo.js'),
    NewMongoIdPlaceholder = require('seed_data/NewMongoIdPlaceholder.js');

module.exports = [
    {
        _id: new NewMongoIdPlaceholder('email_variantFTBSOD_En_Welcome_8AM'),
        name: "8AM Welcome Email",
        due_time_utc: 28800,
        condition: [
            {
                "condition": "country_in",
                "countries": ["us", "ca"],
            }
        ],
        description: "8AM Welcome Email for only those people in us and ca(used in unit tests!)",
        _event_id: new ReferenceTo("ftbsodWelcomeEvent"),
        email_template_slug: "email-schedule-test-template",
        is_default: false,
        send_at_utc_time: false,
        send_from_lead_owner: true,
    },
    {
        _id: new NewMongoIdPlaceholder('email_variantFTBSOD_En_Welcome_1PM'),
        name: "1PM Welcome Email",
        due_time_utc: 39600,
        condition:  [
            {
                "condition": "info_requested",
            }
        ],
        description: "1PM Welcome Email for only those who asked for more info(used in unit tests!)",
        _event_id: new ReferenceTo("ftbsodWelcomeEvent"),
        email_template_slug: "email-schedule-test-template",
        is_default: false,
        send_at_utc_time: false,
        send_from_lead_owner: true,
    },
    {
        _id: new NewMongoIdPlaceholder('email_variantFTBSOD_En_Week_1_Review_8AM'),
        name: "Week 1 Review Email (8 AM)",
        due_time_utc: 28800,
        condition: [],
        description: "La la la, I am a happy placeholder",
        _event_id: new ReferenceTo("ftbsodWeek1ReviewEvent"),
        email_template_slug: "email-schedule-test-template",
        is_default: false,
        send_at_utc_time: false,
        send_from_lead_owner: true,
    },
    {
        _id: new NewMongoIdPlaceholder('email_variantFTBSOD_En_Week_1_Review_1PM'),
        name: "Week 1 Review (1 PM)",
        due_time_utc: 39600,
        condition: [],
        description: "La la la, I am a happy placeholder",
        _event_id: new ReferenceTo("ftbsodWeek1ReviewEvent"),
        email_template_slug: "email-schedule-test-template",
        is_default: true,
        send_at_utc_time: false,
        send_from_lead_owner: true,
    },
];
