// seed_data/objects/courses/fast_track_business_success/foundation/_ftbsod_aus_email_series.js
var ReferenceTo = require('seed_data/ReferenceTo.js'),
    NewMongoIdPlaceholder = require('seed_data/NewMongoIdPlaceholder.js');

module.exports = [
    {
        _id: new NewMongoIdPlaceholder('email_variantFTBSOD_Aus_Welcome_8AM'),
        name: "8AM Welcome Email",
        due_time_utc: 28800,
        condition: [],
        description: "La la la, I am a happy placeholder",
        _event_id: new ReferenceTo("ftbsodWelcomeEvent"),
        email_template_slug: "email-schedule-test-template",
        is_default: false,
        send_at_utc_time: false,
        send_from_lead_owner: true,
    },
    {
        _id: new NewMongoIdPlaceholder('email_variantFTBSOD_Aus_Welcome_1PM'),
        name: "1PM Welcome Email",
        due_time_utc: 39600,
        condition: [],
        description: "La la la, I am a happy placeholder",
        _event_id: new ReferenceTo("ftbsodWelcomeEvent"),
        email_template_slug: "email-schedule-test-template",
        is_default: false,
        send_at_utc_time: false,
        send_from_lead_owner: true,
    },
    {
        _id: new NewMongoIdPlaceholder('email_variantFTBSOD_Aus_Week_1_Review_8AM'),
        name: "Week 1 Review Email (8 AM)",
        due_time_utc: 28800,
        condition: [],
        description: "La la la, I am a happy placeholder",
        _event_id: new ReferenceTo("ftbsodWeek1ReviewEvent"),
        email_template_slug: "email-schedule-test-template",
        is_default: false,
        send_at_utc_time: false,
        send_from_lead_owner: true,
    },
    {
        _id: new NewMongoIdPlaceholder('email_variantFTBSOD_Aus_Week_1_Review_1PM'),
        name: "Week 1 Review Email (1 PM)",
        due_time_utc: 39600,
        condition: [],
        description: "La la la, I am a happy placeholder",
        _event_id: new ReferenceTo("ftbsodWeek1ReviewEvent"),
        email_template_slug: "email-schedule-test-template",
        is_default: true,
        send_at_utc_time: false,
        send_from_lead_owner: true,
    },
];
