// seed_data/objects/courses/fast_track_business_success/foundation/_march_lessons.js
var NewMongoIdPlaceholder = require('seed_data/NewMongoIdPlaceholder.js');

module.exports = [
    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_1'),
        start_time: new Date("2015-03-30T19:00:00+00:00"),
        end_time: new Date("2015-03-30T20:00:00+00:00"),
        actual_start_time: new Date("2015-03-30T19:00:00+00:00"),
        actual_end_time: new Date("2015-03-30T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        recording: {
            vimeo_embed_code: "123550723",
        },
        name: "Business Fundamentals – Structuring for success",
        number: 1,
        expected_availability_date: new Date("2015-03-30T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_2'),
        start_time: new Date("2015-04-01T19:00:00+00:00"),
        end_time: new Date("2015-04-01T20:00:00+00:00"),
        actual_start_time: new Date("2015-04-01T19:00:00+00:00"),
        actual_end_time: new Date("2015-04-01T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        recording: {
            vimeo_embed_code: "1844603",
        },
        name: "Foundation Research",
        number: 2,
        expected_availability_date: new Date("2015-03-31T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_3'),
        start_time: new Date("2015-04-03T19:00:00+00:00"),
        end_time: new Date("2015-04-03T20:00:00+00:00"),
        actual_start_time: new Date("2015-04-03T19:00:00+00:00"),
        actual_end_time: new Date("2015-04-03T20:00:00+00:00"),
        kind: "static",
        extension: false,
        recording: {
            vimeo_embed_code: "123550723",
        },
        name: "Business Effectiveness",
        number: 3,
        expected_availability_date: new Date("2015-04-01T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_4'),
        start_time: new Date("2015-04-06T19:00:00+00:00"),
        end_time: new Date("2015-04-06T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Free Marketing",
        number: 4,
        expected_availability_date: new Date("2015-04-02T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_5'),
        start_time: new Date("2015-04-08T19:00:00+00:00"),
        end_time: new Date("2015-04-08T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Negotiating to a win/win",
        number: 5,
        expected_availability_date: new Date("2015-04-03T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_6'),
        kind: "static",
        extension: false,
        name: "Successful and Fearless Presentations",
        number: 6,
        expected_availability_date: new Date("2015-04-04T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_7'),
        start_time: new Date("2015-04-13T19:00:00+00:00"),
        end_time: new Date("2015-04-13T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "We’re All In Sales",
        number: 7,
        expected_availability_date: new Date("2015-04-05T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_8'),
        start_time: new Date("2015-04-15T19:00:00+00:00"),
        end_time: new Date("2015-04-15T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Closing Sales",
        number: 8,
        expected_availability_date: new Date("2015-04-06T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_9'),
        start_time: new Date("2015-04-20T19:00:00+00:00"),
        end_time: new Date("2015-04-20T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Precision Pricing",
        number: 9,
        expected_availability_date: new Date("2015-04-07T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder('lessonFTBSODMarch_15_10'),
        start_time: new Date("2015-04-22T19:00:00+00:00"),
        end_time: new Date("2015-04-22T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Remaining Focussed",
        number: 10,
        expected_availability_date: new Date("2015-04-08T00:00:00+00:00"),
    },
];
