// seed_data/objects/courses/fast_track_business_success/foundation/call_infos.js
var ReferenceTo = require('seed_data/ReferenceTo.js'),
    ids = require('seed_data/idsToBakeIn.js'),
    ObjectID = require('mongodb').ObjectID;

module.exports = {
    test1CallInfoWelcomeCallFTBS: {
        collection: "CallInfo",
        data: {
            _registration_id: new ReferenceTo("test1TheRegistrationFTBSMarch"),
            _call_type_id: new ReferenceTo("welcomeCallType"),
            date: new Date("2015-03-29T19:00:00+00:00"),
            time_of_day: 'early',
            valid_number: true,
            answered:  true,
            notes: "I had a chat with Test 1",
            _caller_id: new ReferenceTo("jackTheLeadOwner")
        },
    },
    test1CallInfoIntroCallFTBS: {
        collection: "CallInfo",
        data: {
            _registration_id: new ReferenceTo("test1TheRegistrationFTBSMarch"),
            _call_type_id: new ReferenceTo("introCallType"),
            date: new Date("2015-03-30T19:00:00+00:00"),
            time_of_day: 'early',
            valid_number: true,
            answered:  false,
            notes: "I did not have a chat with Test 1",
            _caller_id: new ReferenceTo("jackTheLeadOwner")
        },
    },
    test1CallInfoMidCallFTBS: {
        collection: "CallInfo",
        data: {
            _registration_id: new ReferenceTo("test1TheRegistrationFTBSMarch"),
            _call_type_id: new ReferenceTo("midCallType"),
            date: new Date("2015-04-01T19:00:00+00:00"),
            time_of_day: 'early',
            valid_number: true,
            answered:  false,
            notes: "I did not have a chat with Test 1",
            _caller_id: new ReferenceTo("jackTheLeadOwner")
        },
    },
    test2CallInfoWelcomeCallFTBS: {
        collection: "CallInfo",
        data: {
            _registration_id: new ReferenceTo("test2TheRegistrationFTBSMarch"),
            _call_type_id: new ReferenceTo("welcomeCallType"),
            date: new Date("2015-05-29T19:00:00+00:00"),
            time_of_day: 'early',
            valid_number: false,
            answered:  false,
            notes: "Invalid number",
            _caller_id: new ReferenceTo("jackTheLeadOwner")
        },
    },
};
