// seed_data/objects/courses/fast_track_business_success/foundation/call_emails.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    callEmailFTBS_ENWelcomeAnswered: {
        collection: "CallEmail",
        data: {
            _product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            email_template_slug: 'aft-ftbs-en-welcome-answered',
            filters: {
                _call_type_id: new ReferenceTo("welcomeCallType"),
                valid_number: true,
                answered: true,
                info_requested: null
            }
        }
    },
    callEmailFTBS_ENIntroNoAnswer: {
        collection: "CallEmail",
        data: {
            _product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            email_template_slug: 'aft-ftbs-en-intro-noanswer',
            filters: {
                _call_type_id: new ReferenceTo("introCallType"),
                valid_number: true,
                answered: false,
                info_requested: null
            }
        },
    },
    callEmailFTBS_ENMidBadNumber: {
        collection: "CallEmail",
        data: {
            _product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            email_template_slug: 'aft-ftbs-en-mid-badnumber',
            filters: {
                _call_type_id: new ReferenceTo("midCallType"),
                valid_number: false,
                answered: false,
                info_requested: null
            }
        }
    },
    callEmailFTBS_ENMoreInfo: {
        collection: "CallEmail",
        data: {
            _product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            email_template_slug: 'aft-ftbs-en-more-info',
            filters: {
                _call_type_id: null,
                valid_number: null,
                answered: null,
                info_requested: true
            }
        }
    },
};
