// seed_data/objects/courses/fast_track_business_success/foundation/foundation.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    fastTrackBusinessSuccessOnlineDiplomaCourse: {
        collection: "Course",
        data: {
            name: "Fast Track Business Success Online Diploma",
            domain: "app.shawacademy.com",
            survey: [
                {
                    _question_ids: [
                        new ReferenceTo('trading_q1'),
                        new ReferenceTo('trading_q2'),
                        new ReferenceTo('trading_q3'),
                        new ReferenceTo('trading_q4'),
                        new ReferenceTo('trading_q5'),
                        new ReferenceTo('trading_q6'),
                        new ReferenceTo('trading_q7'),
                        new ReferenceTo('trading_q8'),
                        new ReferenceTo('trading_q9'),
                        new ReferenceTo('trading_q10'),
                        new ReferenceTo('trading_q11'),
                    ]
                }
            ],
            logo: '/images/product_logos/business',
            course_color: '#C2744B',
            slug: 'business',
        }
    },

    ftbsodWelcomeEvent: {
        collection: "Event",
        data: {
            name: "Welcome",
            number: 1,
            _course_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCourse"),
        }
    },

    ftbsodWeek1ReviewEvent: {
        collection: "Event",
        data: {
            name: "Week 1 Review",
            number: 2,
            _course_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCourse"),
        }
    },

    fastTrackBusinessSuccessOnlineDiplomaInEnglish: {
        collection: "Product",
        data: {
            name: "Fast Track Business Success Online Diploma",
            slug: "fast-track-business-success",
            short_slug: "FTBS-EN",
            _course_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCourse"),
            _language_id: new ReferenceTo("english"),
            _linked_product_id: new ReferenceTo("ultimateFastTrackBusinessSuccessInEnglish"),
            stream_names: [
                {
                    name: "ftbs-en-london",
                    timezone: "Europe/London"
                }
            ],
            ultimate: false,
            payment_options: [],
            quizzes: [
                {
                    url: 'http://www.proprofs.com/quiz-school/story.php?title=entrepreneurial-studieslesson1',
                    name: 'Test 1',
                    unlocked_after_lesson_number: 3
                },
                {
                    url: 'http://www.proprofs.com/quiz-school/story.php?title=entrepreneurial-studiesweek2',
                    name: 'Test 2',
                    unlocked_after_lesson_number: 6
                },
                {
                    url: 'http://www.proprofs.com/quiz-school/story.php?title=entrepreneurial-studieslesson2',
                    name: 'Test 3',
                    unlocked_after_lesson_number: 8
                },
                {
                    url: 'http://www.proprofs.com/quiz-school/story.php?title=final-assignment5',
                    name: 'Final Test',
                    unlocked_after_lesson_number: 10
                }
            ],
            resources: [],
        }
    },

    ftbsodEnDefaultEmailSeries: {
        collection: "EmailSeries",
        data: {
            name: "FTBSOD-EN-Default-Email-Series",
            _product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            email_variants: require('./_ftbsod_en_default_email_series.js'),
        }
    },

    ftbsodAusDefaultEmailSeries: {
        collection: "EmailSeries",
        data: {
            name: "FTBSOD-AUS-Default-Email-Series",
            _product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            email_variants: require('./_ftbsod_aus_email_series.js'),
        }
    },

    fastTrackBusinessSuccessOnlineDiplomaCampaignMarch15: {
        collection: "Campaign",
        data: {
            name: "Fast Track Business Success Online Diploma",
            start_date: new Date("2015-03-30T00:00:00+00:00"),
            end_date: new Date("2015-04-22T00:00:00+00:00"),
            offer_end_date: new Date("2015-04-15T00:00:00+00:00"),
            _product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            lessons: require('./_march_lessons.js'),
            timezone: "Europe/London"
        }
    },

    fastTrackBusinessSuccessOnlineDiplomaCampaignMay15: {
        collection: "Campaign",
        data: {
            name: "Fast Track Business Success Online Diploma",
            start_date: new Date("2015-05-04T00:00:00+00:00"),
            end_date: new Date("2015-05-27T00:00:00+00:00"),
            offer_end_date: new Date("2015-05-20T00:00:00+00:00"),
            _product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            lessons: require('./_may_lessons.js'),
            timezone: "Europe/London"
        }
    },
};
