// seed_data/objects/courses/fast_track_business_success/foundation/registrations.js
var ReferenceTo = require('seed_data/ReferenceTo.js'),
    ids = require('seed_data/idsToBakeIn.js'),
    NewMongoIdPlaceholder = require('seed_data/NewMongoIdPlaceholder.js'),
    ObjectID = require('mongodb').ObjectID;


module.exports = {
    fredTheRegistrationFTBS: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("fredThePerson"),
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMay15"),
            attendance: [],
            notes: []
        },
    },
    jimTheRegistrationFTBS: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("jimThePerson"),
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMay15"),
            attendance: [],
            notes: [],
            info_requested: true
        },
    },
    sueTheRegistrationFTBS: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("sueThePerson"),
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMay15"),
            attendance: [],
            notes: []
        },
    },
    samTheRegistrationFTBS: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("samThePerson"),
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMay15"),
            attendance: [],
            notes: []
        },
    },
    jessTheRegistrationFTBS: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("jessThePerson"),
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMay15"),
            attendance: [],
            notes: []
        },
    },
    test1TheRegistrationFTBSMarch: {
        collection: "Registration",
        data: {
            _id: new ObjectID(ids.test1TheRegistration),
            _person_id: new ReferenceTo("test1ThePerson"),
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMarch15"),
            attendance: [],
            notes: [],
            info_requested: true,
            info_sent: false,
            one_to_one: true,
            opt_out: true,
            rating: 'hot'
        },
    },
    test2TheRegistrationFTBSMarch: {
        collection: "Registration",
        data: {
            _id: new ObjectID(ids.test2TheRegistration),
            _person_id: new ReferenceTo("test2ThePerson"),
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMarch15"),
            attendance: [
                {
                    _id: new NewMongoIdPlaceholder('test2TheAttendeeFTBSODMarch_15_1'),
                    _lesson_id: new ObjectID(ids.lessonFTBSODMarch_15_1) ,
                    chat_username: "Test 2",
                    firebase_access_token: "firebase_access_token",
                    live_messages: [
                        {
                            name: "Test 2",
                            message: "Hello",
                            timestamp: 1431366098292
                        },
                        {
                            name: "Test 2",
                            message: "When does it start?",
                            timestamp: 1431366098300
                        },
                        {
                            name: "Support",
                            message: "In 5 mins",
                            timestamp: 1431366098400
                        }

                    ],
                    live_attendance_stats: {
                        total_minutes: 37,
                        active_minutes: 30,
                        idle_minutes: 5,
                        hidden_minutes: 2
                    },
                    recording_attendance_stats: {
                        total_minutes: 23
                    }
                },
                {
                    _id: new NewMongoIdPlaceholder('test2TheAttendeeFTBSODMarch_15_2'),
                    _lesson_id: new ObjectID(ids.lessonFTBSODMarch_15_2) ,
                    chat_username: "Test 2",
                    firebase_access_token: "firebase_access_token",
                    live_messages: [
                        {
                            name: "Test 2",
                            message: "Hi!",
                            timestamp: 1431366098292
                        },
                        {
                            name: "Test 2",
                            message: "LOL",
                            timestamp: 1431366098300
                        },
                        {
                            name: "Support",
                            message: "lol",
                            timestamp: 1431366098400
                        }

                    ],
                    live_attendance_stats: {
                        total_minutes: 60,
                        active_minutes:32,
                        idle_minutes:0,
                        hidden_minutes:28
                    },
                    recording_attendance_stats: {
                        total_minutes: 156
                    }
                },
                {
                    _id: new NewMongoIdPlaceholder('test2TheAttendeeFTBSODMarch_15_3'),
                    _lesson_id: new ObjectID(ids.lessonFTBSODMarch_15_3) ,
                    chat_username: "Test 2",
                    firebase_access_token: "firebase_access_token",
                    live_attendance_stats: {
                        total_minutes: 100,
                        active_minutes:23,
                        idle_minutes:8,
                        hidden_minutes:69,
                    },
                    recording_attendance_stats: {
                        total_minutes: 0
                    }
                },
                {
                    _id: new NewMongoIdPlaceholder('test2TheAttendeeFTBSODMarch_15_4'),
                    _lesson_id: new ObjectID(ids.lessonFTBSODMarch_15_4) ,
                    chat_username: "Test 2",
                    firebase_access_token: "firebase_access_token",
                    live_attendance_stats: {
                    },
                    recording_attendance_stats: {
                        total_minutes: 0
                    }
                },
                {
                    _id: new NewMongoIdPlaceholder('test2TheAttendeeFTBSODMarch_15_5'),
                    _lesson_id: new ObjectID(ids.lessonFTBSODMarch_15_5) ,
                    chat_username: "Test 2",
                    firebase_access_token: "firebase_access_token",
                    live_attendance_stats: {
                        total_minutes: 3,
                        active_minutes:1,
                        idle_minutes:1,
                        hidden_minutes:1,
                    },
                    recording_attendance_stats: {
                        total_minutes: 9
                    }
                }
            ],
            notes: []
        },
    },
    test3TheRegistrationFTBSMarch: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("test3ThePerson"),
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMarch15"),
            attendance: [
                {
                    _id: new NewMongoIdPlaceholder('test3TheAttendeeFTBSODMarch_15_1'),
                    _lesson_id: new ObjectID(ids.lessonFTBSODMarch_15_1) ,
                    chat_username: "Test 3",
                    firebase_access_token: "firebase_access_token",
                    live_attendance_stats: {
                        total_minutes: 50,
                        active_minutes:25,
                        idle_minutes:5,
                        hidden_minutes:20,
                    },
                    recording_attendance_stats: {
                        total_minutes: 23
                    }
                },
                {
                    _id: new NewMongoIdPlaceholder('test3TheAttendeeFTBSODMarch_15_2'),
                    _lesson_id: new ObjectID(ids.lessonFTBSODMarch_15_2) ,
                    chat_username: "Test 3",
                    firebase_access_token: "firebase_access_token",
                    live_attendance_stats: {
                        total_minutes: 60,
                        active_minutes: 20,
                        idle_minutes: 0,
                        hidden_minutes: 40,
                    },
                    recording_attendance_stats: {
                        total_minutes: 156
                    }
                },
                {
                    _id: new NewMongoIdPlaceholder('test3TheAttendeeFTBSODMarch_15_3'),
                    _lesson_id: new ObjectID(ids.lessonFTBSODMarch_15_3) ,
                    chat_username: "Test 3",
                    firebase_access_token: "firebase_access_token",
                    live_attendance_stats: {
                    },
                    recording_attendance_stats: {
                        total_minutes: 89
                    }
                }
            ],
            notes: []
        },
    },
};
