// seed_data/objects/courses/fast_track_business_success/ultimate/_april_lessons.js
var ReferenceTo = require('seed_data/ReferenceTo.js'),
    NewMongoIdPlaceholder = require('seed_data/NewMongoIdPlaceholder.js');

module.exports = [
    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-04-30T19:00:00+00:00"),
        end_time: new Date("2015-04-30T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Business Fundamentals – Structuring for success",
        number: 1,
        expected_availability_date: new Date("2015-04-30T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-05-01T19:00:00+00:00"),
        end_time: new Date("2015-05-01T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Foundation Research",
        number: 2,
        expected_availability_date: new Date("2015-05-01T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        kind: "static",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Business Effectiveness",
        number: 3,
        expected_availability_date: new Date("2015-05-01T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-05-06T19:00:00+00:00"),
        end_time: new Date("2015-05-06T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Free Marketing",
        number: 4,
        expected_availability_date: new Date("2015-05-02T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-05-08T19:00:00+00:00"),
        end_time: new Date("2015-05-08T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Negotiating to a win/win",
        number: 5,
        expected_availability_date: new Date("2015-05-03T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        kind: "static",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Successful and Fearless Presentations",
        number: 6,
        expected_availability_date: new Date("2015-05-04T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-05-13T19:00:00+00:00"),
        end_time: new Date("2015-05-13T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "We’re All In Sales",
        number: 7,
        expected_availability_date: new Date("2015-05-05T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-05-15T19:00:00+00:00"),
        end_time: new Date("2015-05-15T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Closing Sales",
        number: 8,
        expected_availability_date: new Date("2015-05-06T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-05-20T19:00:00+00:00"),
        end_time: new Date("2015-05-20T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Precision Pricing",
        number: 9,
        expected_availability_date: new Date("2015-05-07T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-05-22T19:00:00+00:00"),
        end_time: new Date("2015-05-22T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignApril15"),
        name: "Remaining Focussed",
        number: 10,
        expected_availability_date: new Date("2015-05-08T00:00:00+00:00"),
    },
];
