// seed_data/objects/courses/fast_track_business_success/ultimate/_june_lessons.js
var ReferenceTo = require('seed_data/ReferenceTo.js'),
    NewMongoIdPlaceholder = require('seed_data/NewMongoIdPlaceholder.js');

module.exports = [
    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-06-04T19:00:00+00:00"),
        end_time: new Date("2015-06-04T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Business Fundamentals – Structuring for success",
        number: 1,
        expected_availability_date: new Date("2015-06-30T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-06-06T19:00:00+00:00"),
        end_time: new Date("2015-06-06T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Foundation Research",
        number: 2,
        expected_availability_date: new Date("2015-07-01T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        kind: "static",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Business Effectiveness",
        number: 3,
        expected_availability_date: new Date("2015-07-01T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-06-11T19:00:00+00:00"),
        end_time: new Date("2015-06-11T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Free Marketing",
        number: 4,
        expected_availability_date: new Date("2015-07-02T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-06-13T19:00:00+00:00"),
        end_time: new Date("2015-06-13T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Negotiating to a win/win",
        number: 5,
        expected_availability_date: new Date("2015-07-03T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        kind: "static",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Successful and Fearless Presentations",
        number: 6,
        expected_availability_date: new Date("2015-07-04T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-06-18T19:00:00+00:00"),
        end_time: new Date("2015-06-18T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "We’re All In Sales",
        number: 7,
        expected_availability_date: new Date("2015-07-05T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-06-20T19:00:00+00:00"),
        end_time: new Date("2015-06-20T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Closing Sales",
        number: 8,
        expected_availability_date: new Date("2015-07-06T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-06-25T19:00:00+00:00"),
        end_time: new Date("2015-06-25T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Precision Pricing",
        number: 9,
        expected_availability_date: new Date("2015-07-07T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-06-27T19:00:00+00:00"),
        end_time: new Date("2015-06-27T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        campaign_id: new ReferenceTo("ultimateFastTrackBusinessSuccessCampaignJune15"),
        name: "Remaining Focussed",
        number: 10,
        expected_availability_date: new Date("2015-07-08T00:00:00+0000"),
    },
];
