// seed_data/objects/courses/fast_track_business_success/ultimate/ultimate.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    ultimateFastTrackBusinessSuccess: {
        collection: "Course",
        data: {
            name: "Ultimate Entrepreneurship Program",
            domain: "app.shawacademy.com",
            survey: [],
            logo: '/images/product_logos/business',
            course_color: '#C2744B',
            slug: 'ultimate-business',
        }
    },

    ultimateFastTrackBusinessSuccessInEnglish: {
        collection: "Product",
        data: {
            name: "Ultimate Entrepreneurship Program",
            slug: "ultimate-entrepreneurship",
            short_slug: "UEP-EN",
            _course_id: new ReferenceTo("ultimateFastTrackBusinessSuccess"),
            _language_id: new ReferenceTo("english"),
            _linked_product_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaInEnglish"),
            stream_names: [
                {
                    name: "uep-en-london",
                    timezone: "Europe/London"
                }
            ],
            quizzes: [],
            resources: [],
            ultimate: true,
            payment_options: [
                {
                    type: 'single_course',
                    original_price: 215000,
                    currency: "GBP",
                    country: "gb",
                    instalments: {
                        "1": 59500,
                        "2": 65000,
                        "4": 70000
                    }
                },
                {
                    type: 'single_course',
                    original_price: 215000,
                    currency: "USD",
                    country: "us",
                    instalments: {
                        "1": 89500,
                        "2": 95000,
                        "4": 100000
                    }
                },
                {
                    type: 'single_course',
                    original_price: 215000,
                    currency: "AUD",
                    country: "au",
                    instalments: {
                        "1": 99500,
                        "2": 105000,
                        "4": 110000
                    }
                },
                {
                    type: 'single_course',
                    original_price: 215000,
                    currency: "CAD",
                    country: "ca",
                    instalments: {
                        "1": 94500,
                        "2": 100000,
                        "4": 106000
                    }
                },
                {
                    type: 'single_course',
                    original_price: 215000,
                    currency: "INR",
                    country: "in",
                    instalments: {
                        "1": 1500000,
                        "2": 1700000,
                        "4": 1900000
                    }
                },
                {
                    type: 'single_course',
                    original_price: 215000,
                    currency: "EUR",
                    country: null,
                    instalments: {
                        "1": 69500,
                        "2": 75000,
                        "4": 80000
                    }
                }
            ]
        }
    },

    ultimateFastTrackBusinessSuccessCampaignApril15: {
        collection: "Campaign",
        data: {
            name: "Ultimate Entrepreneurship Program",
            start_date: new Date("2015-04-30T00:00:00+00:00"),
            end_date: new Date("2015-05-22T00:00:00+00:00"),
            _product_id: new ReferenceTo("ultimateFastTrackBusinessSuccessInEnglish"),
            lessons: require('./_april_lessons.js'),
            timezone: "Europe/London"
        },
    },
    ultimateFastTrackBusinessSuccessCampaignJune15: {
        collection: "Campaign",
        data: {
            name: "Ultimate Entrepreneurship Program",
            start_date: new Date("2015-06-04T00:00:00+00:00"),
            end_date: new Date("2015-06-27T00:00:00+00:00"),
            _product_id: new ReferenceTo("ultimateFastTrackBusinessSuccessInEnglish"),
            lessons: require('./_june_lessons.js'),
            timezone: "Europe/London"
        }
    }
};
