// seed_data/objects/courses/trading/foundation/_january_lessons.js
var NewMongoIdPlaceholder = require('seed_data/NewMongoIdPlaceholder.js');

module.exports = [
    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-01-30T19:00:00+00:00"),
        end_time: new Date("2015-01-30T20:00:00+00:00"),
        actual_start_time: new Date("2015-01-30T19:00:00+00:00"),
        actual_end_time: new Date("2015-01-30T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        recording: {
            vimeo_embed_code: "123550723",
        },
        name: "Introduction",
        number: 0,
        expected_availability_date: new Date("2015-01-30T00:00:00+0000"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-01T19:00:00+00:00"),
        end_time: new Date("2015-02-01T20:00:00+00:00"),
        actual_start_time: new Date("2015-02-01T19:00:00+00:00"),
        actual_end_time: new Date("2015-02-01T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        recording: {
            vimeo_embed_code: "1844603",
        },
        name: "The Foundation of Your Foundation Trading Programme",
        number: 1,
        expected_availability_date: new Date("2015-01-31T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-03T19:00:00+00:00"),
        end_time: new Date("2015-02-03T20:00:00+00:00"),
        actual_start_time: new Date("2015-02-03T19:00:00+00:00"),
        actual_end_time: new Date("2015-02-03T20:00:00+00:00"),
        kind: "static",
        extension: false,
        name: "The Mechanics of Trading",
        number: 2,
        expected_availability_date: new Date("2015-02-01T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-06T19:00:00+00:00"),
        end_time: new Date("2015-02-06T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Market Analysis Methodologies",
        number: 3,
        expected_availability_date: new Date("2015-02-02T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-08T19:00:00+00:00"),
        end_time: new Date("2015-02-08T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Candlesticks - Shedding Light on Opportunities",
        number: 4,
        expected_availability_date: new Date("2015-02-03T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        kind: "static",
        extension: false,
        name: "Technical Charting 101",
        number: 5,
        expected_availability_date: new Date("2015-02-04T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-13T19:00:00+00:00"),
        end_time: new Date("2015-02-13T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Building Your Trading Toolkit",
        number: 6,
        expected_availability_date: new Date("2015-02-05T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-15T19:00:00+00:00"),
        end_time: new Date("2015-02-15T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Practice Trading In the Real World - Pros and Cons",
        number: 7,
        expected_availability_date: new Date("2015-02-06T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-20T19:00:00+00:00"),
        end_time: new Date("2015-02-20T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Becoming a Counter Retail Trader",
        number: 8,
        expected_availability_date: new Date("2015-02-07T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-22T19:00:00+00:00"),
        end_time: new Date("2015-02-22T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "Remaining Focussed",
        number: 9,
        expected_availability_date: new Date("2015-02-08T00:00:00+00:00"),
    },

    {
        _id: new NewMongoIdPlaceholder(),
        start_time: new Date("2015-02-22T19:00:00+00:00"),
        end_time: new Date("2015-02-22T20:00:00+00:00"),
        kind: "webinar",
        extension: false,
        name: "The Final Steps - the Practicalities of Real World Trading",
        number: 10,
        expected_availability_date: new Date("2015-02-08T00:00:00+00:00"),
    },
];
