// seed_data/objects/courses/trading/foundation/foundation.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    foundationTradingCourse: {
        collection: "Course",
        data: {
            name: "Foundation Trading Programme",
            domain: "learn.academyft.com",
            survey: [],
            logo: '/images/product_logos/trading',
            course_color: '#333C47',
            slug: 'trading',
        }
    },

    foundationTradingProduct: {
        collection: "Product",
        data: {
            name: "Foundation Trading Programme",
            slug: "foundation-trading-programme",
            short_slug: "utp-en", // This is BS but we need it for tests that use email template slug as we don't have any that match any of our email templates in test at the moment.
            _course_id: new ReferenceTo("foundationTradingCourse"),
            _language_id: new ReferenceTo("english"),
            _linked_product_id: null,
            stream_names: [
                {
                    name: "ftp-london",
                    timezone: "Europe/London"
                }
            ],
            ultimate: false,
            quizzes: [],
            resources: [],
            payment_options: []
        }
    },

    foundationTradingCampaign: {
        collection: "Campaign",
        data: {
            name: "Foundation Trading Programme",
            start_date: new Date("2015-01-01T00:00:00+00:00"),
            end_date: new Date("2015-02-01T00:00:00+00:00"),
            offer_end_date: new Date("2015-02-15T00:00:00+00:00"),
            _product_id: new ReferenceTo("foundationTradingProduct"),
            lessons: require('./_january_lessons.js'),
            timezone: "Europe/London"
        }
    }
};
