// seed_data/objects/courses/digital_marketing/foundation/foundation.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    digitalMarketingCourse: {
        collection: "Course",
        data: {
            name: "Digital Marketing",
            domain: "app.shawacademy.com",
            survey: [],
            logo: '/images/product_logos/digital_marketing',
            course_color: '#AD4943',
            slug: 'marketing',
        }
    },

    digitalMarketingInEnglish: {
        collection: "Product",
        data: {
            name: "Digital Marketing",
            slug: "digital-marketing",
            short_slug: "DM-EN",
            _course_id: new ReferenceTo("digitalMarketingCourse"),
            _language_id: new ReferenceTo("english"),
            _linked_product_id: null,
            stream_names: [
                {
                    name: "dm-en-london",
                    timezone: "Europe/London"
                }
            ],
            ultimate: false,
            quizzes: [],
            resources: [],
            payment_options: []
        }
    },
    digitalMarketingInFrench: {
        collection: "Product",
        data: {
            name: "Marketing en Ligne",
            slug: "marketing-en-ligne",
            short_slug: "DM-FR",
            _course_id: new ReferenceTo("digitalMarketingCourse"),
            _language_id: new ReferenceTo("french"),
            _linked_product_id: null,
            stream_names: [
                {
                    name: "dm-fr-london",
                    timezone: "Europe/London"
                }
            ],
            ultimate: false,
            quizzes: [],
            resources: [],
            payment_options: []
        }
    },

    firstV1DigitalMarketingCampaign: {
        collection: "Campaign",
        data: {
            name: "Diploma in Online Marketing Jan V1",
            start_date: new Date("2015-01-01T00:00:00+00:00"),
            end_date: new Date("2015-02-01T00:00:00+00:00"),
            offer_end_date: new Date("2015-01-20T00:00:00+00:00"),
            _product_id: new ReferenceTo("digitalMarketingInEnglish"),
            lessons: [],
            timezone: "Europe/London"
        }
    },
    firstV2DigitalMarketingCampaign: {
        collection: "Campaign",
        data: {
            name: "Diploma in Online Marketing Jan V2",
            start_date: new Date("2015-01-01T12:00:00+00:00"),
            end_date: new Date("2015-02-01T12:00:00+00:00"),
            offer_end_date: new Date("2015-01-20T00:00:00+00:00"),
            _product_id: new ReferenceTo("digitalMarketingInEnglish"),
            lessons: [],
            timezone: "Europe/London"
        }
    },
    secondDigitalMarketingCampaign: {
        collection: "Campaign",
        data: {
            name: "Diploma in Online Marketing Feb",
            start_date: new Date("2015-02-01T00:00:00+00:00"),
            end_date: new Date("2015-03-01T00:00:00+00:00"),
            offer_end_date: new Date("2015-02-20T00:00:00+00:00"),
            _product_id: new ReferenceTo("digitalMarketingInEnglish"),
            lessons: [],
            timezone: "Europe/London"
        }
    },
};
