// seed_data/objects/courses/digital_marketing/foundation/registrations.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    fredTheRegistration: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("fredThePerson"),
            _campaign_id: new ReferenceTo("hippogriffTamingCampaign"),
            attendance: [],
            notes: []
        },
    },
    jimTheRegistration: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("jimThePerson"),
            _campaign_id: new ReferenceTo("hippogriffTamingCampaign"),
            attendance: [],
            notes: []
        },
    },
    sueTheRegistration: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("sueThePerson"),
            _campaign_id: new ReferenceTo("hippogriffTamingCampaign"),
            attendance: [],
            notes: []
        },
    },
    samTheRegistration: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("samThePerson"),
            _campaign_id: new ReferenceTo("hippogriffTamingCampaign"),
            attendance: [],
            notes: []
        },
    },
    jessTheRegistration: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("jessThePerson"),
            _campaign_id: new ReferenceTo("hippogriffTamingCampaign"),
            attendance: [],
            notes: []
        },
    },
    billTheRegistrationDM1: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("billThePerson"),
            _campaign_id: new ReferenceTo("firstV1DigitalMarketingCampaign"),
            attendance: [],
            notes: []
        },
    },
    fredTheRegistrationDM1: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("fredThePerson"),
            _campaign_id: new ReferenceTo("firstV1DigitalMarketingCampaign"),
            attendance: [],
            notes: []
        },
    },
    jimTheRegistrationDM1: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("jimThePerson"),
            _campaign_id: new ReferenceTo("firstV1DigitalMarketingCampaign"),
            attendance: [],
            notes: []
        },
    },
    sueTheRegistrationDM1: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("sueThePerson"),
            _campaign_id: new ReferenceTo("firstV1DigitalMarketingCampaign"),
            attendance: [],
            notes: []
        },
    },
    samTheRegistrationDM2: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("samThePerson"),
            _campaign_id: new ReferenceTo("firstV2DigitalMarketingCampaign"),
            attendance: [],
            notes: []
        },
    },
    jessTheRegistrationDM2: {
        collection: "Registration",
        data: {
            _person_id: new ReferenceTo("jessThePerson"),
            _campaign_id: new ReferenceTo("firstV2DigitalMarketingCampaign"),
            attendance: [],
            notes: []
        },
    },
};
