// seed_data/objects/purchases/purchases.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    ultimateFastTrackBusinessSuccessPurchase: {
        collection: "Purchase",
        data: {
            _product_id: new ReferenceTo("ultimateFastTrackBusinessSuccess"),
            _person_id: new ReferenceTo("fredThePerson"),
            xero_invoice_id: "whatever",
            stripe_subscription_id: "bla",
            paid: true,
            instalments: 1,
            purchase_date: new Date("2015-05-22T00:00:00+00:00"),
            _registration_id: null,
            is_subscription: false,
            payments: [{
                payment_amount: {
                    currency_code: 'GBP',
                    amount: 10000
                },
                stripe_charge_id: 'foo',
                converted_amount: {
                    currency_code: 'EUR',
                    amount: 9000,
                },
                net_converted_amount: {
                    currency_code: 'EUR',
                    amount: 8500
                },
                is_predicted: false
            }],
            price: {
                currency_code: 'GBP',
                amount: 10000
            },
        }
    },
};
