// seed_data/objects/vouchers/groupon.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    groupon: {
        collection: "Partner",
        data: {
            name: "Groupon",
        }
    },

    grouponDigitalMarketingContract: {
        collection: "PartnerContract",
        data: {
            _partner_id: new ReferenceTo("groupon"),
            price: {
                amount: 9999,
                currency_code: 'GBP'
            }
        }
    },
    grouponHippogriffTamingContract: {
        collection: "PartnerContract",
        data: {
            _partner_id: new ReferenceTo("groupon"),
            Price__c: {
                amount: 4999,
                currency_code: 'GBP'
            }
        }
    },
    fastTrackBusinessSuccessContract: {
        collection: "PartnerContract",
        data: {
            _partner_id: new ReferenceTo("groupon"),
            Price__c: {
                amount: 6999,
                currency_code: 'GBP'
            }
        }
    },

    digitalMarketingVoucher: {
        collection: "Voucher",
        data: {
            _id: "GRP99388404899",
            _course_id: new ReferenceTo("digitalMarketingCourse"),
            redemption_code: "GRP99388404899",
            _partner_contract_id: new ReferenceTo("grouponDigitalMarketingContract"),
        }
    },
    hippogriffTamingVoucher: {
        collection: "Voucher",
        data: {
            _id: "GRP123456HIPPO",
            _course_id: new ReferenceTo("hippogriffTamingCourse"),
            redemption_code: "GRP123456HIPPO",
            _partner_contract_id: new ReferenceTo("grouponHippogriffTamingContract"),
        }
    },
    fastTrackBusinessSuccessVoucher: {
        collection: "Voucher",
        data: {
            _id: "GRP123456FTBS",
            _course_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCourse"),
            redemption_code: "GRP123456FTBS",
            _partner_contract_id: new ReferenceTo("fastTrackBusinessSuccessContract"),
        }
    }
};
