// seed_data/objects/vouchers/redemptions.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    validHippogriffTamingRedemption: {
        collection: "VoucherRedemption",
        data: {
            _person_id: new ReferenceTo("fredThePerson"),
            redeemed: false,
            _voucher_id: new ReferenceTo("hippogriffTamingVoucher"),
        }
    },
    invalidHippogriffTamingRedemption: {
        collection: "VoucherRedemptionError",
        data: {
            _person_id: new ReferenceTo("fredThePerson"),
            _course_id: new ReferenceTo("hippogriffTamingCourse"),
            redeemed: false,
            redemption_code: "SOME-SILLY-NONSENSE-THE-USER-TYPED",
            partner_clue: "Jim's Chilly Voucher Peddler",
            country_code: "AQ",
        }
    },
    validDigitalMarketingRedemption1: {
        collection: "VoucherRedemption",
        data: {
            _person_id: new ReferenceTo("fredThePerson"),
            redeemed: false,
            _voucher_id: new ReferenceTo("digitalMarketingVoucher"),
        }
    },
    validDigitalMarketingRedemption2: {
        collection: "VoucherRedemption",
        data: {
            _person_id: new ReferenceTo("jimThePerson"),
            redeemed: false,
            _voucher_id: new ReferenceTo("digitalMarketingVoucher"),
        }
    },
    validDigitalMarketingRedemption3: {
        collection: "VoucherRedemption",
        data: {
            _person_id: new ReferenceTo("sueThePerson"),
            redeemed: false,
            _voucher_id: new ReferenceTo("digitalMarketingVoucher"),
        }
    },
    validDigitalMarketingRedemption4: {
        collection: "VoucherRedemption",
        data: {
            _person_id: new ReferenceTo("samThePerson"),
            redeemed: false,
            _voucher_id: new ReferenceTo("digitalMarketingVoucher"),
        }
    },
    validDigitalMarketingRedemption5: {
        collection: "VoucherRedemption",
        data: {
            _person_id: new ReferenceTo("jessThePerson"),
            redeemed: false,
            _voucher_id: new ReferenceTo("digitalMarketingVoucher"),
        }
    },


    validFastTrackBusinessSuccessRedemption1: {
        collection: "VoucherRedemption",
        data: {
            _person_id: new ReferenceTo("thePersonWhoIsNotSignedUpToAnythingYet"),
            redeemed: false,
            _voucher_id: new ReferenceTo("fastTrackBusinessSuccessVoucher"),
        }
    },
    invalidHippogriffTamingRedemption1: {
        collection: "VoucherRedemptionError",
        data: {
            _person_id: new ReferenceTo("thePersonWhoIsNotSignedUpToAnythingYet"),
            _course_id: new ReferenceTo("hippogriffTamingCourse"),
            redeemed: false,
            redemption_code: "SOME-SILLY-NONSENSE-THE-USER-TYPED",
            partner_clue: "Jim's Chilly Voucher Peddler",
            country_code: "AQ",
        }
    },
};
