// seed_data/objects/courses/hippogriff_taming/foundation/call_emails.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    callEmailHG_FRIntroAnswered: {
        collection: "CallEmail",
        data: {
            _product_id: new ReferenceTo("hippogriffTamingInFrench"),
            email_template_slug: 'aft-hg-fr-intro-answered',
            filters: {
                _call_type_id: new ReferenceTo("introCallType"),
                valid_number: true,
                answered: true,
                info_requested: null
            }
        }
    },
    callEmailHG_FRMidNoAnswer: {
        collection: "CallEmail",
        data: {
            _product_id: new ReferenceTo("hippogriffTamingInFrench"),
            email_template_slug: 'aft-hg-fr-mid-noanswer',
            filters: {
                _call_type_id: new ReferenceTo("midCallType"),
                valid_number: true,
                answered: false,
                info_requested: null
            }
        }
    },
    callEmailHG_FRMoreInfo: {
        collection: "CallEmail",
        data: {
            _product_id: new ReferenceTo("hippogriffTamingInFrench"),
            email_template_slug: 'aft-hg-fr-more-info',
            filters: {
                _call_type_id: null,
                valid_number: null,
                answered: null,
                info_requested: true
            }
        }
    },
    callEmailHG_FRWelcomeBadNumber: {
        collection: "CallEmail",
        data: {
            _product_id: new ReferenceTo("hippogriffTamingInFrench"),
            email_template_slug: 'aft-hg-fr-welcome-badnumber',
            filters: {
                _call_type_id: new ReferenceTo("welcomeCallType"),
                valid_number: false,
                answered: false,
                info_requested: null
            }
        }
    }
};
