// seed_data/objects/courses/hippogriff_taming/foundation/foundation.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    hippogriffTamingCourse: {
        collection: "Course",
        data: {
            name: "Hippogriff Taming",
            domain: "app.shawacademy.com",
            survey: [],
            logo: '/images/product_logos/hippogriff',
            course_color: '#3699B4',
            slug: 'hippogriffs',
        }
    },

    hippogriffTamingInFrench: {
        collection: "Product",
        data: {
            name: "Hippogriffe Apprivoisement",
            slug: "hippogriffe-apprivoisement",
            short_slug: "HG-FR",
            _course_id: new ReferenceTo("hippogriffTamingCourse"),
            _language_id: new ReferenceTo("french"),
            _linked_product_id: null,
            stream_names: [
                {
                    name: "hg-fr-london",
                    timezone: "Europe/London"
                }
            ],
            ultimate: false,
            quizzes: [],
            resources: [],
            payment_options: []
        }
    },

    hippogriffTamingCampaign: {
        collection: "Campaign",
        data: {
            name: "Hippogriffe Apprivoisement",
            start_date: new Date("2015-06-01T00:00:00+00:00"),
            end_date: new Date("2015-07-01T00:00:00+00:00"),
            offer_end_date: new Date("2015-06-15T00:00:00+00:00"),
            _product_id: new ReferenceTo("hippogriffTamingInFrench"),
            lessons: [],
            timezone: "Europe/London"
        }
    }
};
