// seed_data/objects/people/thePeopleWhoHaveNotSignedUpToAnythingYet.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    test1ThePerson: {
        collection: "Person",
        data: {
            name: {
                title: "Mr",
                first_name: "Test",
                last_name: "1"
            },
            address: {
                country: "gb",
            },
            email: "piptest1@mailinator.com",
            phone: "0987654321",
            lead_info: {
                _lead_owner_id: new ReferenceTo("jillTheLeadOwner")
            },
            evaluation: [],
            is_in_shaw_subscription: false,
            is_in_aft_subscription: false,
        }
    },
    test2ThePerson: {
        collection: "Person",
        data: {
            name: {
                title: "Mr",
                first_name: "Test",
                last_name: "2"
            },
            address: {
                country: "gb",
            },
            email: "piptest2@mailinator.com",
            phone: "00447825210062",
            lead_info: {
                _lead_owner_id: new ReferenceTo("jackTheLeadOwner")
            },
            evaluation: [
                {
                    _question_id: new ReferenceTo("FTBSOD-q1"),
                    answer_value: "female"
                },
                {
                    _question_id: new ReferenceTo("FTBSOD-q2"),
                    answer_value: "18-21"
                },
                {
                    _question_id: new ReferenceTo("FTBSOD-q3"),
                    answer_value: "Account Management"
                },
                {
                    _question_id: new ReferenceTo("FTBSOD-q4"),
                    answer_value: "Intermediate"
                },
                {
                    _question_id:  new ReferenceTo("FTBSOD-q5"),
                    answer_value: "1-3years"
                },
                {
                    _question_id:  new ReferenceTo("FTBSOD-q6"),
                    answer_value: "Curiosity"
                },
                {
                    _question_id:  new ReferenceTo("FTBSOD-q7"),
                    answer_value: "Master Level"
                },
                {
                    _question_id:  new ReferenceTo("FTBSOD-q8"),
                    answer_value: "Unemployed"
                },
                {
                    _question_id:  new ReferenceTo("FTBSOD-q9"),
                    answer_value: "1pm-5pm"
                }
            ],
            is_in_shaw_subscription: false,
            is_in_aft_subscription: false,
        }
    },
    test3ThePerson: {
        collection: "Person",
        data: {
            name: {
                title: "Mr",
                first_name: "Test",
                last_name: "3"
            },
            address: {
                country: "gb",
            },
            email: "piptest3@mailinator.com",
            phone: "0987654321",
            lead_info: {
                _lead_owner_id: null
            },
            evaluation: [],
            is_in_shaw_subscription: false,
            is_in_aft_subscription: false,
        }
    },
};
