// seed_data/objects/email/email_schedules.js
var ReferenceTo = require('seed_data/ReferenceTo.js');

module.exports = {
    welcomeEmailTemplate: {
        collection: "EmailTemplate",
        data: {
            name: "Welcome Email Template",
            vars: ["Course", "Dates", "Info"],
            local_template_ids: [
                {
                    _language_id: new ReferenceTo("english"),
                    template_id: "welcome-email",
                },
            ],
        }
    },

    onTheDayEmailTemplate: {
        collection: "EmailTemplate",
        data: {
            name: "On The Day Template",
            vars: ["Course", "Dates", "Info"],
            local_template_ids: [
                {
                    _language_id: new ReferenceTo("english"),
                    template_id: "on-the-day",
                },
            ],
        }
    },
    successfulMarketerTemplate: {
        collection: "EmailTemplate",
        data: {
            name: "Successful Marketer Template",
            vars: ["Course", "Dates", "Info"],
            local_template_ids: [
                {
                    _language_id: new ReferenceTo("english"),
                    template_id: "successful-marketer",
                },
            ],
        }
    },

    cvtWelcomeEmailDM2PM: {
        collection: "EmailSchedule",
        data: {
            name: "Welcome Email DM 2PM",
            Condition__c: "AND Lead__r.Status = 'Open'",
            due_time_utc: 14 * 60 * 60,
            _email_template_id: new ReferenceTo("welcomeEmailTemplate"),
            _language_id: new ReferenceTo("english"),
            scheduled_emails: [],
        }
    },
    cvtWelcomeEmailDM6PM: {
        collection: "EmailSchedule",
        data: {
            name: "Welcome Email DM 6PM",
            Condition__c: "",
            due_time_utc: 18 * 60 * 60,
            _email_template_id: new ReferenceTo("welcomeEmailTemplate"),
            _language_id: new ReferenceTo("english"),
            scheduled_emails: [],
        }
    },
    cvtOnTheDayEmailDM: {
        collection: "EmailSchedule",
        data: {
            name: "On The Day Email",
            Condition__c: "AND Lead__r.Status = 'Open'",
            due_time_utc: 09 * 60 * 60,
            _email_template_id: new ReferenceTo("onTheDayEmailTemplate"),
            _language_id: new ReferenceTo("english"),
            scheduled_emails: [],
        }
    },
    cvtSuccessfulMarketer1EmailDM: {
        collection: "EmailSchedule",
        data: {
            name: "Successful Marketer 1 Email DM",
            Condition__c: "AND Lead__r.Status = 'Open'",
            due_time_utc: 09 * 60 * 60,
            _email_template_id: new ReferenceTo("successfulMarketerTemplate"),
            _language_id: new ReferenceTo("english"),
            scheduled_emails: [],
        }
    },
    cvtSuccessfulMarketer2EmailDM: {
        collection: "EmailSchedule",
        data: {
            name: "Successful Marketer 2 Email DM",
            Condition__c: "AND Lead__r.Status = 'Open'",
            due_time_utc: 09 * 60 * 60,
            _email_template_id: new ReferenceTo("successfulMarketerTemplate"),
            _language_id: new ReferenceTo("english"),
            scheduled_emails: [],
        }
    }
};
