// seed_data/objects/email/scheduled_email.js
var ReferenceTo = require('seed_data/ReferenceTo.js'),
    ids = require('seed_data/idsToBakeIn.js'),
    ObjectID = require('mongodb').ObjectID;

module.exports = {
    ftbsodEnWelcomeScheduledEmail: {
        collection: "ScheduledEmail",
        data: {
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMay15"),
            _email_variant_ids: [
                ObjectID(ids.email_variantFTBSOD_En_Welcome_8AM),
                ObjectID(ids.email_variantFTBSOD_En_Welcome_1PM)
            ],
            due_date_utc: new Date("2015-05-04T00:00:00+00:00"),
        }
    },
    ftbsodEnWeek1ReviewScheduledEmail: {
        collection: "ScheduledEmail",
        data: {
            _campaign_id: new ReferenceTo("fastTrackBusinessSuccessOnlineDiplomaCampaignMay15"),
            _email_variant_ids: [
                ObjectID(ids.email_variantFTBSOD_En_Week_1_Review_8AM),
                ObjectID(ids.email_variantFTBSOD_En_Week_1_Review_1PM)
            ],
            due_date_utc: new Date("2015-05-05T00:00:00+00:00"),
        }
    },
    ftbsodSueEnWeek1ReviewScheduledEmailReceipt: {
        collection: "ScheduledEmailReceipt",
        data: {
            _scheduled_email_id: new ReferenceTo("ftbsodEnWelcomeScheduledEmail"),
            _email_variant_id: ObjectID(ids.email_variantFTBSOD_En_Welcome_8AM),
            sent_date: new Date("2015-05-05T00:00:00+00:00"),
            registration_id: new ReferenceTo("sueTheRegistrationFTBS")
        }
    },
};
