// seed_data/objects/calls/call_types.js
module.exports = {
    welcomeCallType: {
        collection: "CallType",
        data: {
            name: "Welcome",
            number: 1,
        }
    },
    introCallType: {
        collection: "CallType",
        data: {
            name: "Intro",
            number: 2,
        }
    },
    midCallType: {
        collection: "CallType",
        data: {
            name: "Mid",
            number: 3,
        }
    },
    sales1CallType: {
        collection: "CallType",
        data: {
            name: "Sales 1",
            number: 4,
        }
    },
    sales2CallType: {
        collection: "CallType",
        data: {
            name: "Sales 2",
            number: 5,
        }
    },
    sales3CallType: {
        collection: "CallType",
        data: {
            name: "Sales 3",
            number: 6,
        }
    },
    sales4CallType: {
        collection: "CallType",
        data: {
            name: "Sales 4",
            number: 7,
        }
    },
    sales5CallType: {
        collection: "CallType",
        data: {
            name: "Sales 5",
            number: 8,
        }
    },

};
